### PR TITLE
Use client function to unify `client` in test script

### PR DESCRIPTION
**Pull request by bananaappletw**
I recreated the pull request.
cc: @darrachequesne

### CODE REVIEW

File: test/socket.io.js [hunk 1]
```diff
@@ -1546,9 +1546,7 @@ describe('socket.io', function(){
       var srv = http();
       var sio = io(srv);
       srv.listen(function() {
-        var addr = srv.listen().address();
-        var url = 'ws://localhost:' + addr.port + '?key1=1&key2=2';
-        var socket = ioc(url);
+        var socket = client(srv, {query: {key1: 1, key2: 2}});
```
Comment (darrachequesne) [inline test/socket.io.js]: Isn't the meaning of the test updated there?

Comment (bananaappletw) [inline test/socket.io.js]: Nope.
Actually, this is the same meaning as the previous test.
This pull request just simplify the code.

Comment (bananaappletw) [inline test/socket.io.js]: https://github.com/socketio/socket.io-client/blob/master/lib/index.js#L44
In here, the query is parsed from the url.
`key1=1&key2=2`
You could simply add debug code in socket.io-client to see the value of `opts.query`.
Eventually, they are parsed as the same value.

Comment (bananaappletw) [inline test/socket.io.js]: @darrachequesne

Comment (bananaappletw) [inline test/socket.io.js]: According this:
https://github.com/socketio/socket.io/blob/master/.github/PULL_REQUEST_TEMPLATE.md
[Improve code quality]

**Comment by darrachequesne** [comment]
@bananaappletw thanks!!

**Pull request merged**